### ISSUE TITLE
No way to add an accessible name or an aria-role to an image

### ISSUE THREAD

**Issue by fstorr** [nygardk/react-share#258]
These issues are necessary to improve the accessibility of the share icons.

Each icon is an image, so it needs to be possible to add a [`role="img"` attribute](https://www.w3.org/TR/wai-aria-1.1/#img) so that assistive technology (e.g. a screen reader) will announce the icon as an image.

There also must be a way to add either an [`aria-label`](https://www.w3.org/TR/wai-aria-1.1/#aria-label) or an [`aria-labelledby`](https://www.w3.org/TR/wai-aria-1.1/#aria-labelledby) attribute to the element. The `aria-label` attribute takes a string (e.g. "Twitter", "Twitter logo") and the `aria-labelledby` attribute takes a reference to one or more space-separated `id` attributes.

An example could be:
```
<TwitterIcon role={img} ariaLabel={"Twitter"} … />
```

which would transform to:
```
<svg role="img" aria-label="Twitter" … >
  …
</svg>
```

@jaronheard

**Comment by jaronheard** [comment]
@nygardk Would you be willing to accept a PR for this issue?

Here's what props I would propose adding:

| aria attribute | prop  | defaultProp | propType |
| ------------- | ------------- | ------------- | ------------- |
| role  | role  | "img"  | String |
| aria-label  | ariaLabel  | network (e.g. "Twitter")  | String |
| aria-hidden  | ariaHidden  | true  | Boolean |

I'd add these in the IconFactory.js file.

For most applications, the accessibility would be much improved with out the implementing developer doing anything, but you'd still have the option to override the values for special circumstances.

Let me know if you'd like anything done differently!

**Comment by nygardk** [comment]
Fixed in `v4.0.0`. Icons accept now all `svg` attributes and buttons accept all `button` attributes.
